iron-checked-element-behavior.html

```diff
@@ -0,0 +1,75 @@
+<!--
+@license
+Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../iron-validatable-behavior/iron-validatable-behavior.html">
+
+
+<script>
+
+  /**
+   * Use `Polymer.IronCheckedElementBehavior` to implement a custom element
+   * that has a `checked` property, which can be used for validation if the
+   * element is also `required`. Element instances implementing this behavior
+   * will also be registered for use in an `iron-form` element.
+   *
+   * @demo demo/index.html
+   * @polymerBehavior Polymer.IronCheckedElementBehavior
+   */
+  Polymer.IronCheckedElementBehaviorImpl = {
+
+    properties: {
+
+      /**
+       * Gets or sets the state, `true` is checked and `false` is unchecked.
+       */
+      checked: {
+        type: Boolean,
+        value: false,
+        reflectToAttribute: true,
+        notify: true,
+      },
+
+      /**
+       * If true, the button toggles the active state with each tap or press
+       * of the spacebar.
+       */
+      toggles: {
+        type: Boolean,
+        value: true,
+        reflectToAttribute: true
+      },
+
+      /**
+       * Set to true to mark the input as required.
+       */
+      required: {
```

Comment:
> I think we should consider our spaghetti bowl of behaviors and input elements more carefully and figure out where these shared attributes fit in.

Generally this sounds like a good idea. We should do it.

For now, some hopefully useful Devil's advocacy: in the case of `core-focusable`, I took the approach of separating it into its core domain concepts. In the long run, my opinion is that this made the approach to focusability more conceptually simple at its core, but architecturally more complex when it came to implementing it in other elements.

So, with hindsight, I think that merging concepts into a single behavior isn't necessarily bad for the sake of broader architectural simplicity.

For example, let's say I want to implement a new form element that needs a validation behavior. I apply `IronValidatableBehavior`. Some time passes, and I use the element I made in a form. Some of the other form elements I used also apply `IronValidatableBehavior`, and they are marked as `required`. So, I add a `required` attribute to my element. But wait, it doesn't use `IronRequiredValidatableBehavior`, so `required` doesn't actually do anything.

If I am the element author, I can just go change the behavior hierarchy of the element above. It's annoying, but I can do it. However, if I am not the element author, I am in a bad place and need to submit a PR so that I can use `IronRequiredValidatableBehavior` in the element.

On the one hand, it makes a lot of architectural sense that `IronRequiredValidatableBehavior` and `IronValidatableBehavior` are not the same thing. The first one layers additional functionality on the base that is the second one.

On the other hand, I don't know if I would ever want to use `IronValidatableBehavior`, because the `required` attribute is such a generalizable concept for form elements and it would be inconsistent to pick and choose which ones support it.

Anyway, sorry for the essay. WDYT?